README.md

```diff
@@ -26,23 +26,39 @@ Please set explicitly your `cluster_version` to an older EKS version until your
 A full example leveraging other community modules is contained in the [examples/basic directory](https://github.com/terraform-aws-modules/terraform-aws-eks/tree/master/examples/basic).
 
 ```hcl
-data "aws_eks_cluster" "cluster" {
-  name = module.my-cluster.cluster_id
+module "my-cluster" {
+  source          = "terraform-aws-modules/eks/aws"
+  cluster_name    = "my-cluster"
+  cluster_version = "1.16"
+  subnets         = ["subnet-abcde012", "subnet-bcde012a", "subnet-fghi345a"]
+  vpc_id          = "vpc-1234556abcdef"
+
+  worker_groups = [
+    {
+      instance_type = "m4.large"
+      asg_max_size  = 5
+    }
+  ]
 }
+```
+
+## Creating Kubernetes resources in the same plan
 
+The module has a kubernetes provider embedded for managing some resources. This is not accessible to the calling module. You will need to define your own kubernetes provider block. The data source is necessary so that the kubernetes provider waits for the cluster to be created before attempting to communicate with it:
+```hcl
 data "aws_eks_cluster_auth" "cluster" {
-  name = module.my-cluster.cluster_id
+  name = module.eks.cluster_id
```

Comment:
You only really need the one data source to block initialization of the kubernetes provider.

Reply:
yes.